Readability.js

```diff
@@ -847,7 +847,7 @@ Readability.prototype = {
 
         // Initialize and score ancestors.
         this._forEachNode(ancestors, function(ancestor, level) {
-          if (!ancestor.tagName)
+          if (!ancestor.tagName || !ancestor.parentNode || typeof(ancestor.parentNode.tagName) === 'undefined')
```

Comment:
Do we really need that last check? :-)

Reply:
Not sure, I just copied it from the check on `elementToScore` a couple dozen lines earlier